### PR TITLE
Prevent null values of unit_cost by Stock Entry form

### LINKED ISSUE #4718: Stock Integration can set unit_cost to 0
It turns out that I can set the `unit_cost` to NULL during the stock integration.  This results in a database error.  It would be more useful to catch this on the client-side.  See below.

GJGXkf1Dhw

### PR DESCRIPTION
**Pull request by jmcameron**
Prevents deleting unit cost field and make sure that it is a number > 0.    
Fixes #4718.

Suggested testing procedure:
  * Use the sequence outlined in Issue #4718 
  * Try deleting the Unit Cost field.  The modal dialog should report an error and prevent submission.
  * Verify that angular prevents entry of non-numbers
  * Check entering negative value

Translation:
  * Please review French translation of STOCK.ERRORS.MISSING_LOT_UNIT_COST

### CODE REVIEW

File: client/src/modules/stock/entry/modals/StockEntryModalForm.service.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ StockEntryModalForm.$inject = ['uuid'];
 function StockEntryModalForm(uuid) {
   const ERR_NO_ROWS = 'STOCK.ERRORS.NO_ROWS';
   const ERR_LOT_QUANTITY_OVER_GLOBAL = 'STOCK.ERRORS.LOT_QUANTITY_OVER_GLOBAL';
+  const ERR_MISSING_LOT_UNIT_COST = 'STOCK.ERRORS.MISSING_LOT_UNIT_COST';
```
Comment (jniles) [inline client/src/modules/stock/entry/modals/StockEntryModalForm.service.js:17]: Nice 👍

**Review by jniles — APPROVED** [review]
Looks good to me.  ~~This review indicates to me that we don't have unit tests for this form ... and we really need them.  I'll make an issues about it.~~

Verified:
7QTSl8qyfY


bors r+

**Comment by jniles** [comment]
bors r-

Turns out, I didn't look hard enough - we do have unit tests for this form: https://github.com/IMA-WorldHealth/bhima/blob/master/test/client-unit/services/StockEntryModalForm.spec.js

Before merging this, could you add a unit tests that shows the error if the unit cost is set to undefined?  It should look like this:

https://github.com/IMA-WorldHealth/bhima/blob/437130d4f4224a1021195b3cf4c579cafe6e3d1a/test/client-unit/services/StockEntryModalForm.spec.js#L185-L203

File: client/src/i18n/fr/stock.json [hunk 2]
```diff
@@ -42,6 +42,7 @@
       "LOT_ROW_WITH_ERRORS" : "Données invalide dans certaines lignes de lot ",
       "EXCESSIVE_QUANTITY": "Quantité excessive",
       "LOT_QUANTITY_OVER_GLOBAL": "La quantité totale des lots dépasse la quantity globale fournie",
+      "MISSING_LOT_UNIT_COST": "Une valeur non négative est requise pour le coût unitaire du lot",
```
Comment (jniles) [inline client/src/i18n/fr/stock.json:45]: This looks fine to me.

**Comment by jmcameron** [comment]
Extended the unit test for StockForms to catch missing unit_cost.  @jniles

File: test/client-unit/services/StockEntryModalForm.spec.js [hunk 3]
```diff
@@ -192,9 +192,11 @@ describe('StockEntryModalForm', () => {
     third.quantity = -100;
 
     const form = new StockForm({ rows : [first, second, third], expires : true });
+    delete form.unit_cost;
     const errors = form.validate();
 
     const expectedErrors = [
+      'STOCK.ERRORS.MISSING_LOT_UNIT_COST',
```
Comment (jniles) [inline test/client-unit/services/StockEntryModalForm.spec.js:199]: Excellent :+1:

Comment (jniles) [inline test/client-unit/services/StockEntryModalForm.spec.js:199]: Excellent :+1:

**Review by jniles — APPROVED** [review]
This looks good to me!  Sorry about the start and cancellation of the merging process.

bors r+

**Pull request merged**